Production/govuk_ios/Repositories/TopicsRepository.swift

```diff
@@ -0,0 +1,88 @@
+import Foundation
+import CoreData
+
+protocol TopicsRepositoryInterface {
+    func saveTopicsList(_ topicResponses: [TopicResponseItem])
+    func fetchFavoriteTopics() -> [Topic]
+    func fetchAllTopics() -> [Topic]
+    func saveChanges()
+}
+
+struct TopicsRepository: TopicsRepositoryInterface {
+    private let coreData: CoreDataRepositoryInterface
+
+    init(coreData: CoreDataRepositoryInterface) {
+        self.coreData = coreData
+    }
+
+    func saveTopicsList(_ topicResponses: [TopicResponseItem]) {
+        let context = coreData.backgroundContext
+        let isFirstLaunch = fetchAllTopics().count == 0
+        topicResponses.forEach { topicResponse in
+            if !topicExists(for: topicResponse,
+                            in: context) {
+                createTopic(for: topicResponse,
```

Comment:
I might be missing something, but if a topic is updated i.e new title, this isn't going to save the updated title, is it?